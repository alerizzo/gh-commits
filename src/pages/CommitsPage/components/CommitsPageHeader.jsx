import React from 'react';
import { TextInput } from 'components';
import CommitRow from 'components/CommitsList/CommitRow';

export default function CommitsPageHeader({ onSearch, repository, disabled, showTableHeader }) {
  return (
    <div className="CommitsPageHeader">
      <h1 className="title is-3">Commits list</h1>
      <div className="field">
        <label className="label is-size-6">
          Search
          {repository && (
            <span className="has-text-weight-light">
              {' in '}
              {repository.owner}'s {repository.repository} ({repository.refName})
            </span>
          )}
        </label>
        <TextInput style={{ maxWidth: '50%' }} disabled={disabled} placeholder="Search ..." />
      </div>
      {showTableHeader && (
        <div className="CommitsPageHeader_TableHeader">
          <CommitRow.Header />
        </div>
      )}
    </div>
  );
}
