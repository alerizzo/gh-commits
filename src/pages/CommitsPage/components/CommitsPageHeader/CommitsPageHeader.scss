@import '../../../../scss/variables';

.CommitsPageHeader {
  background-color: $grey-lighter;
  padding: $gap;
  position: sticky;
  top: $navbar-height;
  z-index: 1;

  .CommitsPageHeader_TableHeader {
    margin: 0 $gap * -1 $gap * -1;
    padding: $gap $gap 12px;
    background: $white;
    font-size: $size-7;
    text-transform: uppercase;
  }
}
