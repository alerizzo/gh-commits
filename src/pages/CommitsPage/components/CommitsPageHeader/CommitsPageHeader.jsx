import React, { useEffect } from 'react';
import { TextInput } from 'components';
import CommitRow from 'components/CommitsList/CommitRow';
import { useInput } from 'lib/hooks';
import './CommitsPageHeader.scss';

export default function CommitsPageHeader({
  onSearch,
  isSearching,
  repository,
  disabled,
  showTableHeader,
}) {
  const searchInput = useInput('');

  useEffect(() => {
    onSearch && onSearch(searchInput.value);
  }, [searchInput.value, onSearch]);

  return (
    <div className="CommitsPageHeader">
      <h1 className="title is-3">Commits List</h1>
      <div className="field">
        <label className="label is-size-6">
          Search
          {repository && (
            <span className="has-text-weight-light">
              {' in '}
              {repository.owner}'s {repository.repository} ({repository.refName})
            </span>
          )}
        </label>
        <TextInput
          style={{ width: '276px' }}
          disabled={disabled}
          isLoading={isSearching}
          placeholder="Search ..."
          {...searchInput}
        />
      </div>
      {showTableHeader && (
        <div className="CommitsPageHeader_TableHeader">
          <CommitRow.Header />
        </div>
      )}
    </div>
  );
}
