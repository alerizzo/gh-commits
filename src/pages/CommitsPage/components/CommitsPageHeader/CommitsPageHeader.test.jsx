import React from 'react';
import renderer from 'react-test-renderer';

import CommitsPageHeader from './CommitsPageHeader';

describe('<CommitsPageHeader />', () => {
  const defaultProps = {};
  const wrapper = renderer.create(<CommitsPageHeader {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<CommitsPageHeader isSearching />', () => {
  const defaultProps = {
    isSearching: true,
  };
  const wrapper = renderer.create(<CommitsPageHeader {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<CommitsPageHeader repository={...} />', () => {
  const defaultProps = {
    repository: {
      owner: 'Owner',
      repository: 'Repository',
      refName: 'Ref Name',
    },
  };
  const wrapper = renderer.create(<CommitsPageHeader {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<CommitsPageHeader showTableHeader />', () => {
  const defaultProps = {
    showTableHeader: true,
  };
  const wrapper = renderer.create(<CommitsPageHeader {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
