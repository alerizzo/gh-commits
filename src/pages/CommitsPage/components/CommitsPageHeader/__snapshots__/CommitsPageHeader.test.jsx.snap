// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommitsPageHeader /> render 1`] = `
<div
  className="CommitsPageHeader"
>
  <h1
    className="title is-3"
  >
    Commits List
  </h1>
  <div
    className="field"
  >
    <label
      className="label is-size-6"
    >
      Search
    </label>
    <div
      className="TextInput control "
      data-error={null}
      style={
        Object {
          "width": "276px",
        }
      }
    >
      <input
        className="input  "
        onChange={[Function]}
        placeholder="Search ..."
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<CommitsPageHeader isSearching /> render 1`] = `
<div
  className="CommitsPageHeader"
>
  <h1
    className="title is-3"
  >
    Commits List
  </h1>
  <div
    className="field"
  >
    <label
      className="label is-size-6"
    >
      Search
    </label>
    <div
      className="TextInput control is-loading"
      data-error={null}
      style={
        Object {
          "width": "276px",
        }
      }
    >
      <input
        className="input  "
        onChange={[Function]}
        placeholder="Search ..."
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<CommitsPageHeader repository={...} /> render 1`] = `
<div
  className="CommitsPageHeader"
>
  <h1
    className="title is-3"
  >
    Commits List
  </h1>
  <div
    className="field"
  >
    <label
      className="label is-size-6"
    >
      Search
      <span
        className="has-text-weight-light"
      >
         in 
        Owner
        's 
        Repository
         (
        Ref Name
        )
      </span>
    </label>
    <div
      className="TextInput control "
      data-error={null}
      style={
        Object {
          "width": "276px",
        }
      }
    >
      <input
        className="input  "
        onChange={[Function]}
        placeholder="Search ..."
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<CommitsPageHeader showTableHeader /> render 1`] = `
<div
  className="CommitsPageHeader"
>
  <h1
    className="title is-3"
  >
    Commits List
  </h1>
  <div
    className="field"
  >
    <label
      className="label is-size-6"
    >
      Search
    </label>
    <div
      className="TextInput control "
      data-error={null}
      style={
        Object {
          "width": "276px",
        }
      }
    >
      <input
        className="input  "
        onChange={[Function]}
        placeholder="Search ..."
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="CommitsPageHeader_TableHeader"
  >
    <div
      className="columns CommitRow CommitRow_Header"
    >
      <div
        className="column is-2"
      >
        Author
      </div>
      <div
        className="column is-2"
      >
        Commit
      </div>
      <div
        className="column is-5"
      >
        Message
      </div>
      <div
        className="column is-3"
      >
        Created
      </div>
    </div>
  </div>
</div>
`;
